platform/admin/ui/src/main/webapp/js/views/application/IFrameView.js

```diff
@@ -12,18 +12,38 @@
  * <http://www.gnu.org/licenses/lgpl.html>.
  *
  **/
-/* global define */
+/* global define,setInterval,clearInterval */
 define([
     'marionette',
     'icanhaz',
-    'text!iframeView'
-    ],function (Marionette, ich, iframeView) {
+    'text!iframeView',
+    'text!waitForVisible'
+    ],function (Marionette, ich, iframeView, waitForVisible) {
 
     ich.addTemplate('iframeView',iframeView);
+    ich.addTemplate('waitForVisible',waitForVisible);
 
     var IFrameView = Marionette.ItemView.extend({
         template: 'iframeView',
         className: 'iframe-view'
     });
-    return IFrameView;
-});
\ No newline at end of file
+
+    return Marionette.Layout.extend({
+        template: 'waitForVisible',
+        isVisible: function () {
+            return this.$el.is(':visible');
+        },
+        regions: { iframe: '.wait-for-visible' },
+        initialize: function () {
+            // delay rendering of iframe until the element is actually visible
+            var interval = setInterval(function () {
```

Comment:
Have you tried to make this event driven yet?  I think that would be better than polling with an interval.

http://getbootstrap.com/javascript/#tabs-events
http://stackoverflow.com/a/19482691